adapters/AppLovin/AppLovinAdapter/GADMAppLovinMediatedNativeAppInstallAd.m

```diff
@@ -0,0 +1,116 @@
+//
+//  GADMAppLovinMediatedNativeAppInstallAd.m
+//  SDK Network Adapters Test App
+//
+//  Created by Santosh Bagadi on 4/12/18.
+//  Copyright © 2018 AppLovin Corp. All rights reserved.
+//
+
+#import "GADMAppLovinMediatedNativeAppInstallAd.h"
+
+#import "GADMAppLovinExtraAssets.h"
+
+@interface GADMAppLovinMediatedNativeAppInstallAd() <GADMediatedNativeAdDelegate>
+
+@property(nonatomic, strong) ALNativeAd *nativeAd;
+@property(nonatomic, strong) NSArray *nativeAdImages;
+@property(nonatomic, strong) GADNativeAdImage *nativeAdIcon;
+@property(nonatomic, copy) NSDictionary<NSString *, id> *extras;
+
+@end
+
+@implementation GADMAppLovinMediatedNativeAppInstallAd
+
+- (instancetype)initWithNativeAd:(ALNativeAd *)nativeAd {
+    if (!nativeAd)
+    {
+        return nil;
+    }
+    
+    self = [super init];
+    if (self) {
+        self.nativeAd = nativeAd;
+        UIImage *downloadedImage =
+        [UIImage imageWithData: [NSData dataWithContentsOfURL: self.nativeAd.imageURL]];
+        GADNativeAdImage *image = [[GADNativeAdImage alloc] initWithImage: downloadedImage];
+        self.nativeAdImages = @[ image ];
+        UIImage *downloadedIcon =
+        [UIImage imageWithData: [NSData dataWithContentsOfURL: self.nativeAd.iconURL]];
+        self.nativeAdIcon = [[GADNativeAdImage alloc] initWithImage: downloadedIcon];
+        
+        NSMutableDictionary *extraAssets =
+        [NSMutableDictionary dictionaryWithObject:self.nativeAd.adIdNumber forKey:GADMAppLovinAdID];
+        if (self.nativeAd.captionText) {
+            extraAssets[GADMAppLovinCaption] = self.nativeAd.captionText;
+        }
+        self.extras = extraAssets;
+    }
+    return self;
+}
+
+- (NSDictionary * _Nullable)extraAssets {
+    return self.extras;
+}
+
+- (NSString * _Nullable)body {
+    return self.nativeAd.descriptionText;
+}
+
+- (NSString * _Nullable)callToAction {
+    return self.nativeAd.ctaText;
+}
+
+- (NSString * _Nullable)headline {
+    return self.nativeAd.title;
+}
+
+- (GADNativeAdImage * _Nullable)icon {
+    return self.nativeAdIcon;
+}
+
+- (NSArray * _Nullable)images {
+    return self.nativeAdImages;
+}
+
+- (NSString * _Nullable)price {
+    return nil;
+}
+
+- (NSDecimalNumber * _Nullable)starRating {
+    return [[NSDecimalNumber alloc] initWithFloat:self.nativeAd.starRating.floatValue];
+}
+
+- (NSString * _Nullable)store {
+    return nil;
+}
+
+- (UIView *)mediaView {
+    GADNativeAdImage *nativeAdImage = (GADNativeAdImage *)[self.nativeAdImages firstObject];
+    UIImage *image = [nativeAdImage image];
+    UIImageView *mainImageView = [[UIImageView alloc] initWithImage:image];
```

Comment:
Causing "Main Thread Checker: UI API called on a background thread: -[UIImageView initWithImage:]" error because of calling the initWithimage in a background thread.